src/components/access-type.js

```diff
@@ -233,9 +228,65 @@ class AccessTypeBase extends React.Component {
         }
       ]
     };
+    if (success_url && cancel_url) {
+      paymentObject.options = {};
+
+      paymentObject.options.urls = {
+        success_url: success_url,
+        cancel_url: cancel_url
+      };
+    }
+    return paymentObject;
+  }
+
+  initRazorPayPayment = (selectedPlan, planType = "", storyId = "", storyHeadline = "", storySlug = "") => {
+    if (!selectedPlan) {
+      console.warn("Razor pay needs a plan");
+      return false;
+    }
+
+    const { paymentOptions } = this.props;
+    const paymentType = get(selectedPlan, ["recurring"]) ? "razorpay_recurring" : "razorpay";
+    const paymentObject = this.makePaymentObject(
+      selectedPlan,
+      planType,
+      storyId,
+      storyHeadline,
+      storySlug,
+      paymentType
+    );
     return paymentOptions.razorpay.proceed(paymentObject);
   };
 
+  initStripePayment = (
```

Comment:
We can't keep writing methods for every payment option that AT introduce. instead of `initRazorPayPayment, initStripePayment, initPaypalPayment etc`, lets write one method something like `initPayment` and pass payment gateway name as parameter to method. e.g. `initPayment(RazorPay)`.

This way we don't need work again on component even AT brings new payment gateway tomorrow, unless it requires additional information.

Reply:
Make sense, but we don't know which publisher is using  `initRazorPayPayment` if we change it to some other function like you mentioned that will break. also ahead is using the same thing.